### ISSUE TITLE
Restart will create new build number

### ISSUE THREAD

**Issue by bradrydzewski** [drone/drone#2035]
Restart has gotten much more complex over time. Previously when a build was restarted we just had to reset a few columns and re-run the build. But with 0.6 we now have to update multiple rows across multiple tables (matrix rows, step rows, step log rows, build row)

This complexity has caused me to re-evaluate the current implementation. I believe going forward builds should be immutable. Once a build is complete the results can never be changed.

Restarting a build will therefore create a new build number, and will re-direct the user to a new build results screen.

**Comment by orlando** [comment]
What about Github status in this case, will they work?. For example, we restart builds if they fail due something not related to tests (example, when NPM timeouts). Will restarting the build still work for Github statuses?

**Comment by bradrydzewski** [comment]
that status is tied to the commit sha / commit ref. So creating a new build number would still result in the status being correctly updated in subsequent builds :)

**Comment by orlando** [comment]
@bradrydzewski 👌  thanks for the clarification

**Comment by bradrydzewski** [comment]
pull request ready https://github.com/drone/drone/pull/2059

**Comment by ozbillwang** [comment]
Thanks @bradrydzewski 

Could we also add feature who triggered the restart build as well for easy follow up?

**Comment by bradrydzewski** [comment]
implemented by #2204

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (109/100 (109%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:59209 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drCzX-002evr-VF
	for greg@biomassiv.es; Sun, 10 Sep 2017 19:54:24 -0500
Date: Sun, 10 Sep 2017 17:53:43 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505091223;
	bh=X96lkA5mmhnkowvn2Y908/iN7iBifNrjKwlXZo0Wpm0=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=SNgtuh0yv8a693PSdZc6OR0exEo6QS7P6Fvab0iLTUHmkuG2qta0xw+B1tEo0uUKL
	 xODxwLiZiABwXqel8f/CW4GbczMSkHAroDrpjTq8xHnFS37FW86ypQ9drIPWBnAdPH
	 MHbPGmDlqVq0OyPCfxmOmwi2TkMFQ8ez1Z3zFD68=
